frontend/src/components/elements/DeckGlJsonChart/styled-components.ts

```diff
@@ -30,9 +34,25 @@ export const StyledDeckGlChart = styled.div<StyledDeckGlChartProps>(
   })
 )
 
-export const StyledNavigationControlContainer = styled.div(({ theme }) => ({
-  position: "absolute",
-  right: "2.625rem",
-  top: theme.spacing.md,
-  zIndex: 1,
-}))
+export const StyledNavigationControlContainer =
+  styled.div<StyledNavigationControlContainerProps>(({ theme, mapTheme }) => ({
+    position: "absolute",
+    right: "2.625rem",
+    top: theme.spacing.md,
+    zIndex: 1,
+
+    // Update zoom buttons based on the active theme
+    "button:not(:disabled)": {
+      background: theme.colors.bgColor,
+
+      // Add a separator between buttons
+      "& + button": {
+        borderTopColor: theme.colors.secondaryBg,
+      },
+
+      // On dark backgrounds, invert the color for the + and - symbols
+      "& span": {
+        filter: mapTheme === "dark" ? "invert(100%)" : "",
```

Comment:
nit: have you also thought about having the map buttons in sync with the configured theme (via `hasLightBackgroundColor`) instead of the map theme? Currently, most of the map charts actually don't make use of this dark/light switch on the frontend and instead have a map style configured from the backend.

Reply:
Let me know if what I did on [385cab1](https://github.com/streamlit/streamlit/pull/5479/commits/385cab104ba63af8fff78f1fbef790838ca593a2) addresses this suggestion! I _think_ I was already doing what you were referring to, but in the wrong place haha. Or maybe I'm not understanding your suggestion correctly... 

Also, maybe `mapTheme` is a poor variable naming? Let me know if `activeTheme` would make more sense!

Reply:
I think you can even simplify it more and directly call the light background check inside the styled component, e.g. 

```typescript
filter: hasLightBackgroundColor(theme) ? "invert(100%)" : "",
```

This way you can fully remove the `mapTheme` property from the `DeckObject`.

Reply:
Got it! Updated in [c263e54](https://github.com/streamlit/streamlit/pull/5479/commits/c263e54684122952466acf270b97e6f26398d004)